File: 4057-total-waviness-of-numbers-in-range-i/total-waviness-of-numbers-in-range-i.java
class Solution {
    public int totalWaviness(int num1, int num2) {
        int c = 0;
        for(int i=num1; i<=num2; i++){
            String j = Integer.toString(i);
            for(int k = 1; k < j.length() - 1; k++){
                char first = j.charAt(k);
                char second = j.charAt(k+1);
                char prev = j.charAt(k-1);
                if (first > second && first > prev){
                    c += 1;
                    continue;
                }
                if (first < second && first < prev){
                    c += 1;
                    continue;
                }
            }
        }
        return c;
    }
}

File: 4057-total-waviness-of-numbers-in-range-i/total-waviness-of-numbers-in-range-i.py
class Solution(object):
    def totalWaviness(self, num1, num2):
        count = 0
        for i in range(num1, num2+1):
            i = str(i)
            for j in range(1, len(i)-1):
                
                prev = i[j-1]
                curr = i[j]
                nex = i[j+1]
                if prev < curr > nex or prev > curr < nex:
                    count += 1
        return count


